src/pipx/venv_inspect.py

```diff
@@ -0,0 +1,227 @@
+import json
+import logging
+from pathlib import Path
+from typing import Any, Dict, List, NamedTuple, Optional, Set, Tuple
+
+from packaging.requirements import Requirement
+from packaging.utils import canonicalize_name
+
+try:
+    from importlib import metadata
+except ImportError:
+    import importlib_metadata as metadata  # type: ignore
+
+from pipx.util import PipxError, run_subprocess
+
+try:
+    WindowsError
+except NameError:
+    WINDOWS = False
+else:
+    WINDOWS = True
+
+logger = logging.getLogger(__name__)
+
+
+class VenvInspectInformation(NamedTuple):
+    distributions: List[metadata.Distribution]
+    bin_path: Path
+
+
+class VenvMetadata(NamedTuple):
+    apps: List[str]
+    app_paths: List[Path]
+    apps_of_dependencies: List[str]
+    app_paths_of_dependencies: Dict[str, List[Path]]
+    package_version: str
+    python_version: str
+
+
+def get_dist(
+    package: str, distributions: List[metadata.Distribution]
+) -> Optional[metadata.Distribution]:
+    """Find matching distribution in the canonicalized sense."""
+    for dist in distributions:
+        if canonicalize_name(dist.metadata["name"]) == canonicalize_name(package):
+            return dist
+    return None
+
+
+def get_package_dependencies(
+    dist: metadata.Distribution, extras: Set[Any]
+) -> List[Requirement]:
+    # Add an empty extra to enable evaluation of non-extra markers
+    if not extras:
+        extras.add("")
+    dependencies = []
+    for req in map(Requirement, dist.requires or []):
+        if not req.marker:
+            dependencies.append(req)
+        else:
+            for extra in extras:
+                if req.marker.evaluate({"extra": extra}):
```

Comment:
Darn, @uranusjr's comments in another section just reminded me of this statement.  Unfortunately `req.marker.evaluate` is using the current environment.  When this code was running in the venv it was accurately evaluating environment markers.  But running in pipx it may not be (for example the python version may be wrong.)

Reply:
Ah!  But `Marker.evaluate()` accepts an `environment` dict to supply an override-environment.  I need to implement that.

Reply:
OK, I added implementation of code to get the environment inside the venv to apply when executing `Marker.evaluate()` to determine if a dependency is used.